grapher_admin/views.py

```diff
@@ -503,6 +509,9 @@ def store_import_data(request: HttpRequest):
                     with connection.cursor() as cursor:
                         cursor.execute("DELETE FROM sources WHERE sources.id NOT IN (SELECT variables.sourceId FROM variables)")
 
+                csv_write_thread = Thread(target=write_dataset_csv, args=(dataset.pk, dataset_old_name, request.user.get_full_name(), request.user.email))
```

Comment:
I'd recommend not doing this in a thread. If the thread fails for some reason, the SQL database will be updated but the corresponding commit will be lost, creating inconsistency between the export archive and the grapher output.

Instead, it should be done synchronously within the same transaction block as the rest of the import, so that if it errors out, the SQL changes are also rolled back. It's fine if the import process takes a little while.